Controllers/FamilyController.cs

```diff
@@ -139,47 +241,202 @@ public async Task<IActionResult> CreateFamily([FromForm] FamilyViewModel Family)
+                            Value = r.Text
+                        };
+                    var created = _considerationsService.CreateConsideration(restriction);
+                    if (!created.Success)
+                    {
+                        return Task.FromException(
+                            new Exception($"Error creating consideration. Error: {created.Error}")
+                        );
+                    }
+                }
+            }
+
+            foreach (SelectListItem g in Member.Goals)
+            {
+                if (g.Selected && contextMember != null)
+                {
+                    ConsiderationsCreateDto goal =
+                        new()
+                        {
+                            MemberId = contextMember.MemberId,
+                            Type = ConsiderationsEnum.Goal,
+                            Value = g.Text
+                        };
+                    var created = _considerationsService.CreateConsideration(goal);
+                    if (!created.Success)
+                    {
+                        return Task.FromException(
+                            new Exception($"Error creating consideration. Error: {created.Error}")
+                        );
+                    }
+                }
+            }
+
+            foreach (SelectListItem c in Member.Cuisines)
+            {
+                if (c.Selected && contextMember != null)
+                {
+                    ConsiderationsCreateDto cuisine =
+                        new()
+                        {
+                            MemberId = contextMember.MemberId,
+                            Type = ConsiderationsEnum.Cuisine,
+                            Value = c.Text
+                        };
+                    var created = _considerationsService.CreateConsideration(cuisine);
+                    if (!created.Success)
+                    {
+                        return Task.FromException(
+                            new Exception($"Error creating consideration. Error: {created.Error}")
+                        );
+                    }
+                }
+            }
+            contextMember = null;
+        }
+
+        return Task.CompletedTask;
     }
 
-    [HttpPut("{Id}")]
-    public ActionResult<FamilyModel> UpdateFamily(string Id, [FromBody] FamilyUpdateDto family)
+    // Handles the deletion of considerations if a member was to update theirs
+    private Task DeleteOldConsiderations(string memberId)
     {
-        var updated = _familyService.UpdateFamily(Id, family);
+        var considerations = _considerationsService.GetMemberConsiderations(memberId).Data;
 
-        if (!updated.Success)
+        if (considerations != null)
         {
-            return BadRequest($"Error: {updated.Error}");
-        }
+            var timeFrame = DateTime.UtcNow.AddSeconds(-10);
+            foreach (var consideration in considerations)
+            {
```

Comment:
I added the time frame just in case there was any processing time delay. Seems to be working. Maybe I should decrease it to 5 or so?

Reply:
a timestamp should be taken just before we insert the new considerations. the timestamp could be passed to this function in order to provide the comparison, that way we aren't guessing what the delay will be.

Reply:
ahh, that does make more sense! Good catch!